### PR TITLE
Put the crafting gui code into it's own module.

### PR DESCRIPTION
**Pull request by JornVernee**
Let's shorten crafting.cpp further, and separate logic code from gui code.

I think there's a lot of changes to the gui code that can be made to make it easier to understand/maintain (e.g. splitting the ~300 lines of `select_crafting_recipe` into several functions, so we can at the very least keep track of indentation a little better). I think putting it in it's own module is a good first step.

~~The only thing that's kind of weird right now is that the crafting category and crafting sub-category containers are still used by both files. These containers are a part of the model, not the view (imho). It might make sense to put them in the recipe dictionary module.~~

### CODE REVIEW

File: CataclysmWin.cbp [hunk 1]
```diff
@@ -372,6 +372,8 @@
 		<Unit filename="src/coordinates.h" />
 		<Unit filename="src/crafting.cpp" />
 		<Unit filename="src/crafting.h" />
+		<Unit filename="src/crafting_GUI.cpp" />
```
Comment (kevingranade) [inline CataclysmWin.cbp]: No need to all-caps "gu"i, just crafting_ui.bla would fit in better.

Comment (JornVernee) [inline CataclysmWin.cbp]: yeah, I was going to ask about that :)

**Comment by JornVernee** [comment]
I noticed a bug while doing the last commit: The crafting recipe categories are loaded dynamically from json, but the gui draws from a constant list defined inside the code (in ```draw_recipe_tabs``` and ```draw_recipe_subtabs```).

**Comment by JornVernee** [comment]
hmm, committing the renaming of files is not something I did before, but this works too I guess.

File: src/init.cpp [hunk 2]
```diff
@@ -14,6 +14,7 @@
 #include "item_factory.h"
 #include "vehicle_group.h"
 #include "crafting.h"
+#include "crafting_GUI.h"
```
Comment (kevingranade) [inline src/init.cpp]: missed one.

**Pull request merged**